app/v2/controllers/V2MovementsController.scala

```diff
@@ -533,7 +533,8 @@ class V2MovementsControllerImpl @Inject() (
                  |Reason: ${failureDetails.failureReason}
                  |Message: ${failureDetails.message}""".stripMargin)
 
-            // TODO: Push notification on failure
+            pushNotificationsService
+              .postPpnsNotification(movementId, messageId, Json.toJson(HateoasMovementUpdateResponse(movementId, messageId, movementType, None)))
```

Comment:
@drnaylor-oc  I think we need to send upscan failure details here  UpscanFailedResponse(reference, failureDetails) instead of HateoasMovementUpdateResponse(movementId, messageId, movementType, None)?

Reply:
Yes, a failure shouldn't be returning a Hateoas object. This should just be a bad request presentation error.

Reply:
@olakunlekuye2  Please change this as suggested.

Reply:
Done